lib/classes/checkingaccount_class.rb

```diff
@@ -0,0 +1,47 @@
+module BankAccount
+
+  class CheckingAccount < Account
+    attr_reader :checks
+
+    #inherits passed in parameters from Account class
+    def initialize(id, initial_balance)
+    super(id, initial_balance)
+    @current_balance = initial_balance
+    @checks = 0
+    end
+
+    # used super to inherit functionality for overdraft
+    # may not be the best use of super since still needed the if/else block for the transaction fee
+    # could potentially refactor Account #withdraw to use a variable that accounts for fees/minimum balances
+    # but seems like it may be a little more complicated than needed
```

Comment:
There are a few other approaches you could take to sharing code. For example if you make a method

```ruby
def fee
 0
end
```
On the base `Account` class and the rewrite the `withdraw` conditional like: `if amount > @current_balance - fee` you could override the `fee` method here and completely reuse the withdraw method. That probably isn't worth it in this context, but hopefully it helps show other ways to share code.